packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/extension.tsp

```diff
@@ -11,7 +11,7 @@ using TypeSpec.OpenAPI;
 
 namespace Azure.ResourceManager.Models.Resources;
 
-@resource("customExtensionResources")
+@parentResource(TopLevelTrackedResource)
```

Comment:
We don't need @parentResource, extension resource can have more than 1 parent type.
Just change the mockapi.ts to accept top-level resource as its parent(and in your test for generated code in autorest.java, pass in the top-level resource id), and it should be fine.

Reply:
@XiaofeiCao 
If you don't use `@parentResource`, then there must be a parameter in Autorest.java that accepts the top-level resource id, but I haven't been able to find a `Resource` or `Parameter` that I can use.